test/extended/machines/cluster.go

```diff
@@ -126,46 +124,18 @@ var _ = g.Describe("[sig-node] Managed cluster", func() {
 		actual := make(map[string]int)
 		errs := make([]error, 0)
 
-		// Find a number of rendered-* machineconfigs for masters and workers
-		// Each rendered config adds expected reboot
-		expectedReboots := map[string]int{
-			"worker": 0,
-			"master": 0,
-		}
-		dynamicClient, err := dynamic.NewForConfig(oc.KubeFramework().ClientConfig())
-		o.Expect(err).NotTo(o.HaveOccurred())
-		machineConfigGroupVersionResource := schema.GroupVersionResource{
-			Group: "machineconfiguration.openshift.io", Version: "v1", Resource: "machineconfigs",
-		}
-		mcList, err := dynamicClient.Resource(machineConfigGroupVersionResource).List(ctx, metav1.ListOptions{})
-		if apierrors.IsNotFound(err) {
-			g.Skip("Skipping test because we have no machineconfigs.machineconfiguration.openshift.io.")
-			return
-		}
-		o.Expect(err).NotTo(o.HaveOccurred())
-		for i := range mcList.Items {
-			mcName := mcList.Items[i].GetName()
-			for prefix := range expectedReboots {
-				if strings.HasPrefix(mcName, fmt.Sprintf("rendered-%s-", prefix)) {
-					expectedReboots[prefix] += 1
-				}
-			}
-		}
-
 		// List nodes, set actual and expected number of reboots
 		for _, node := range nodes.Items {
 			// Examine only the nodes which are available at the start of the test
 			if !slices.Contains(staticNodeNames, node.Name) {
 				continue
 			}
-			expectedRebootsForNodeRole := expectedReboots[getNodeRole(&node)]
-			o.Expect(expectedReboots).To(o.HaveKey(expectedRebootsForNodeRole))
-			expected[node.Name] = expectedRebootsForNodeRole
-			nodeReboots, err := getNumberOfBootsForNode(oc.KubeClient(), oc.Namespace(), node.Name)
+			nodeBoots, nodeReboots, err := getNumberOfBootsForNode(oc.KubeClient(), oc.Namespace(), node.Name)
 			if err != nil {
 				errs = append(errs, err)
 			}
-			actual[node.Name] = nodeReboots
+			expected[node.Name] = nodeBoots
+			actual[node.Name] = nodeReboots + 1
```

Comment:
Why this is +1?

Reply:
This compares:
* number of machine boots
* number of legitimate reboots

The former is always larger than the former (the machine boots, gets rebooted, boots again), this is why we add 1 here, so that `expected` would match `actual` later